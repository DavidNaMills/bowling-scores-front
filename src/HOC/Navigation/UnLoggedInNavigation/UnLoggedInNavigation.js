import React from 'react';
import { NavLink } from 'react-router-dom';
import classes from '../Navigation.module.scss';


const UnLoggedInNavigation = (props) => (
    <div className={classes.navigation}>
        <div className={classes.navigation_section}>
            <NavLink exact className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/'>Home</NavLink>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/game'>Game</NavLink>
        </div>
        <div className={classes.navigation_section}>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/login'>Login</NavLink>
            <NavLink className={classes.navigation__link} activeClassName={classes.navigation__link_active} to='/signup'>Signup</NavLink>
        </div>
    </div>
)

export default UnLoggedInNavigation;