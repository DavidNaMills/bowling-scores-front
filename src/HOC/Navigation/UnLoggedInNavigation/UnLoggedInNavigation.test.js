import React from 'react';
import {mount} from 'enzyme';
import {NavLink, BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import UnLoggedInNavigation from './UnLoggedInNavigation';


describe('<UnLoggedInNavigation/> test suite', ()=>{
    it('should match the snapshot', () => {
        const component = renderer.create(
            <BrowserRouter>
                <UnLoggedInNavigation />
            </BrowserRouter>
        );
        const app = component.toJSON();
        expect(app).toMatchSnapshot();
    });

    describe('render tests', () => {
        const wrapper = mount(
            <BrowserRouter>
                <UnLoggedInNavigation />
            </BrowserRouter>
        );

        it('contains 4 NavLink components', () => {
            expect(wrapper.find(NavLink).length).toBe(4);
        });

        it('contains 2 sections', () => {
            expect(wrapper.find('.navigation_section').length).toBe(2);
        });
    });
});