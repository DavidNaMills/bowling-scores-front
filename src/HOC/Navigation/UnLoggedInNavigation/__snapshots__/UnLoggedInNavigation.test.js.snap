// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnLoggedInNavigation/> test suite should match the snapshot 1`] = `
<div
  className="navigation"
>
  <div
    className="navigation_section"
  >
    <a
      aria-current="page"
      className="navigation__link navigation__link_active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="navigation__link"
      href="/game"
      onClick={[Function]}
    >
      Game
    </a>
  </div>
  <div
    className="navigation_section"
  >
    <a
      aria-current={null}
      className="navigation__link"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <a
      aria-current={null}
      className="navigation__link"
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
  </div>
</div>
`;
